streams/src/main/java/org/apache/kafka/streams/state/internals/SegmentedBytesStore.java

```diff
@@ -96,7 +96,7 @@
      */
     byte[] get(Bytes key);
 
-    interface KeySchema {
+    interface KeySchema<S extends Segment> {
```

Comment:
Is Session/WindowKeySchema always templated on KeyValueSegment (seems indicated in the javadoc below at line 157)?

Reply:
Yes. Atm, `KeyValueSegment` is the only class implementing `Segment`.

Reply:
As discussed in person: we can simplify this and remove the generic type from the interface and only use a generic type on method `segmentsToSearch`.